### DISCUSSION TITLE
Tailwind-tools with Django-Tailwind

### DISCUSSION THREAD

**Discussion by l2late** [luckasRanarison/tailwind-tools.nvim#55] [Q&A] [answered]
Hello everyone,

After having worked on a tailwind project in vscode. This plugin made me want to try it out in nvim.
I've been trying to get the plugin to work, unfortunately without success.

I can't find anything useful in the LSP logs. All I noticed is that I can't get the tailwind (and html) LSP to set the root dir correctly in nvim. Other LSPs (python, ruff...) work fine.

Note that I added Tailwind to my Django project using [Django-Tailwind](https://django-tailwind.readthedocs.io/en/latest/installation.html)

Has anyone had a similar issue?
Could anyone give me some pointers as to where I might look for a solution?

Your time and help is greatly appreciated!

**Comment by luckasRanarison** [comment]
Hi, I'm not a Django user but where is the `tailwind.config.*` file located in a Django tailwind project? Currently, I've configured lspconfig in a way that it only launches when it detects a `tailwind.config.*` file at the top of the root directory or inside a folder called `assets` (Phoenix projects), so that could be the issue.

**Comment by l2late** [comment]
Hello Luckas!
Thank you for your answer!

The Django-Tailwind plugin places all tailwind & js related files in a "theme" directory at the project root.
The tailwind.config.js file is itself in <project-root>/theme/static_src/
Below, you'll find an example of the structure of said <project-root>/theme.

├── apps.py
├── static
│   └── css
│       └── dist
│           └── styles.css
├── static_src
│   ├── node_modules/
│   ├── package.json
│   ├── package-lock.json
│   ├── postcss.config.js
│   ├── src/
│   │   └── styles.css
│   └── tailwind.config.js
└── templates/
    └── base.html

If I'm understanding you right, i need to configure a different root-dir for the lsp.
Is there a preferred way to configure a different project root for tailwind-tools?

**Reply by luckasRanarison** [reply, marked as answer]
No, you don't have to add anything, the plugin should be responsible of configuring that.

I've added support for Django tailwind it should work now. Let me know if it doesn't.

**Reply by l2late** [reply]
Thank you! Works like a charm now!
I only couldn't get the colors to preview in cmp but fixed it by manually adding tailwindcss-colorizer-cmp.nvim
